packages/ts-transforms/src/command.ts

```
@@ -121,8 +123,10 @@ function parseData(data: string): object[] | null {
             try {
                 results.push(JSON.parse(line));
             } catch (err) {
-                logger.error(`Failed to parse "${line}"`);
-                return null;
+                if (!ignoreErrors) {
```

If `ignoreErrors === true` is set we should still log it

---

it was never getting logged at all. When -i is set, by default it ignore's parsing errors. That was the intention. If its not set, it throws

---

It was being log using `logger.error(...);`, but since it was returning `null` it would short circuit the parsing. Regardless the error shouldn't go into the abyss even if the `ignoreErrors` is set to `true`

---

That may be the case, but since it would always short circuit then essentially it was doing nothing. Running the cli command normally will produce the appropriate error and point to the offending data. For development purposes (which this is mainly for) if you are working with dirty data that might have issues then you can use the -i on purpose to ignore errors so that it actually can run in the first place and second to not get cluttered with errors and get the response. -i is an unsafe mode so to speak, it's purpose to my knowledge would be to throw them to the abyss on purpose. -i is not intended to be used normally

---

My policy with errors is they should rarely be ignored completely. My interpretation (and expectation) about a feature like this is to allow the CLI to skip over parsing errors but not fail early, but that doesn't mean they shouldn't outputted. If the errors are silently ignored, a developer can't be informed that there was problem parsing and may blame the rules or something else.

But I guess I am not really aware of the requirements for this, maybe @kstaken could provide better input than I.